Update config.md (#321)
returntocorp/sgrep: docs/config.md

## Config format

> This document provides examples for the .yaml format for specifying rules to sgrep. It does not cover the patterns themselves; for [the pattern syntax see patterns.md](patterns.md)

Reviewer: This config doc does more than provide examples: it also defines/describes operators, fields, and their composition. Thus I’d suggest something like:

```suggestion
This document describes sgrep rule fields and provides rule examples. It does not cover the patterns themselves; for [the pattern syntax see patterns.md](patterns.md)
```

I wonder if the content for patterns.md should be merged into this doc? I see that patterns.md is rather long and that might be a challenge, but they seem nearly inseparable (can’t have a rule without a pattern, and you can’t have a pattern without a rule).

Author: @pabloest agreed they should probably be unified eventually, but don't see an easy way to do that now

> If you want run these rules or ship them to your team, rather than write them, check out the easy-to-use [bento.dev](https://bento.dev)

Reviewer: ```suggestion
If instead of writing rules, you want run them, ship them to your team, or enforce them in CI/CD, check out the easy-to-use [bento.dev](https://bento.dev)
```

check my commas here, not quite sure if I got them right

    severity: ERROR
```

> When invoked with `sgrep -f myconfig.yaml project/`, this rule will run on Python files inside `project/` and output the rule `id` "eqeq-is-bad" along with the `message` and the line and column range the match occurred on. Given a line inside a Python file with `1 == 1`, sgrep will output the message: "1 == 1 is always true!"

Reviewer: ```suggestion
When invoked with `sgrep -f myconfig.yaml project/`, this rule will run on Python files inside `project/` and outputs the rule `id` "eqeq-is-bad" along with the `message` and the line and column range the match occurred on. For example, given a line inside a Python file with `1 == 1`, sgrep will output the message: "1 == 1 is always true!"
```

Author: I think this change actually makes it unnecessarily wordier but maybe I'm missing something

Reviewer: Agreed, no need to make it wordier.
